securedrop/tests/migrations/migration_fccf57ceef02.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+
+import random
+import uuid
+
+from sqlalchemy import text
+from sqlalchemy.exc import NoSuchColumnError
+
+from db import db
+from journalist_app import create_app
+from models import Submission
+from .helpers import random_bool, random_chars, random_datetime, bool_or_none
+
+random.seed('ᕕ( ᐛ )ᕗ')
+
+
+def add_source():
+    filesystem_id = random_chars(96) if random_bool() else None
+    params = {
+        'filesystem_id': filesystem_id,
+        'uuid': str(uuid.uuid4()),
+        'journalist_designation': random_chars(50),
+        'flagged': bool_or_none(),
+        'last_updated': random_datetime(nullable=True),
+        'pending': bool_or_none(),
+        'interaction_count': random.randint(0, 1000),
+    }
+    sql = '''INSERT INTO sources (filesystem_id, uuid,
+                journalist_designation, flagged, last_updated, pending,
+                interaction_count)
+             VALUES (:filesystem_id, :uuid, :journalist_designation,
+                :flagged, :last_updated, :pending, :interaction_count)
+          '''
+    db.engine.execute(text(sql), **params)
+
+
+class UpgradeTester():
+
+    '''This migration verifies that the UUID column now exists, and that
+    the data migration completed successfully.
+    '''
+
+    SOURCE_NUM = 200
+
+    def __init__(self, config):
+        self.config = config
+        self.app = create_app(config)
+
+    def load_data(self):
+        with self.app.app_context():
+
+            for _ in range(self.SOURCE_NUM):
+                add_source()
+
+            for sid in range(1, self.SOURCE_NUM, 8):
+                for _ in range(random.randint(1, 3)):
+                    self.add_submission(sid)
+
+            # create "abandoned" submissions (issue #1189)
+            for sid in range(self.SOURCE_NUM, self.SOURCE_NUM + 50):
+                self.add_submission(sid)
+
+            db.session.commit()
+
+    @staticmethod
+    def add_submission(source_id):
+        params = {
+            'source_id': source_id,
+            'filename': random_chars(50),
+            'size': random.randint(0, 1024 * 1024 * 500),
```

Comment:
Nit: like in `qa_loader.py` it may be fine to use this instead to use less disk space:

```python
f_len = int(math.floor(random.expovariate(100000) * 1024 * 1024 * 500))
```